server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2229,7 +2243,9 @@ public void forceMerge(final boolean flush, int maxNumSegments, boolean onlyExpu
                     this.forceMergeUUID = forceMergeUUID;
                 }
                 if (flush) {
-                    flush(false, true);
+                    PlainActionFuture<FlushResult> future = PlainActionFuture.newFuture();
+                    flush(false, true, future);
+                    future.actionGet();
```

Comment:
I think we may want to let force merge become async too in order to not occupy the force merge thread. Hence we might as well rely on the default implementation of `flush(boolean, boolean)` to ensure we come back to this?